File: Excel/300-399/313/313 Challenge.R
library(tidyverse)
library(readxl)

path <- "300-399/313/313 Scan3.xlsx"
input <- read_excel(
  path,
  range = "A2:G5",
  col_names = FALSE,
  col_types = "text"
)
test <- read_excel(path, range = "I2:O5", col_names = FALSE, col_types = "text")

result <- as.data.frame(t(apply(input, 1, function(row) {
  sapply(0:6, function(k) {
    paste(row[seq(k %% 2 + 1, k + 1, by = 2)], collapse = "")
  })
})))

all.equal(unname(result), unname(test))
# [1] TRUE

File: Excel/300-399/313/313 Challenge.py
import pandas as pd

path = "300-399/313/313 Scan3.xlsx"
input_df = pd.read_excel(path, usecols="A:G", skiprows=1, nrows=4, header=None, dtype=str)
test     = pd.read_excel(path, usecols="I:O", skiprows=1, nrows=4, header=None, dtype=str)

result = pd.DataFrame(
    {
        k: input_df.apply(
            lambda row: "".join(row.iloc[list(range(k % 2, k + 1, 2))]),
            axis=1,
        )
        for k in range(7)
    }
).reset_index(drop=True)
result.columns = range(7)

print(result.equals(test))
# True
